package/gluon-config-mode-geo-location/src/0400-geo-location.lua.in

```diff
@@ -0,0 +1,148 @@
+				© <a href="https://openstreetmap.org/copyright">OpenStreetMap</a> contributors
+			</div>
+		</div>
+		<script type="text/javascript" src="/static/osm.js"></script>
+		<script type="text/javascript">
+			var latitude=]] .. show_lon() .. ",longitude=" .. show_lat() .. ",zoom=" .. site_zoom() .. [[;
+			var openLayerSrc = document.createElement("script");
+			openLayerSrc.type = 'text/javascript';
+			openLayerSrc.src = "]] .. show_olurl() .. [[?noCache=]] .. os.time() .. [[";
+			openLayerSrc.onload = openLayerSrc.onreadystatechange = function () {
+				window.addEventListener("load", showMap, false);
+				var lpm=document.getElementById("locationPickerMap");
+				setInterval(function() {
+					!1 !== findObj("longitude") ? lpm.style.display = "block" : lpm.style.display = "none";
+				}, 100);
+			};
+			document.head.appendChild(openLayerSrc);
+		</script>
+		<div>
+	]]
+
+#endif
+
+	local s = form:section(Section, nil, text)
+
+	local uci_share_location = uci:get_bool("gluon-node-info", location, "share_location")
+
+	local geolocation = s:option(ListValue, "geolocation", pkg_i18n.translate("Geo-Location"))
+	geolocation:value("static", pkg_i18n.translate("Static location"))
+	geolocation:value("none", pkg_i18n.translate("Disabled"))
+	if uci_latitude == nil and uci_longitude == nil and uci_share_location == false then
+		geolocation.default = "none"
+	else
+		geolocation.default = "static"
+	end
+
+	local share_location = s:option(Flag, "sharelocation", pkg_i18n.translate("Share your location to see your router on the map"))
+	share_location.default = uci_share_location
+	share_location:depends(geolocation, "static")
+
+	local latitude = s:option(Value, "latitude", pkg_i18n.translate("Latitude"), pkg_i18n.translatef("e.g. %s", "50.364931"))
+	latitude.default = uci_latitude
+	latitude:depends(geolocation, "static")
+	latitude.datatype = "float"
+
+	local longitude = s:option(Value, "longitude", pkg_i18n.translate("Longitude"), pkg_i18n.translatef("e.g. %s", "7.606417"))
+	longitude.default = uci_longitude
+	longitude:depends(geolocation, "static")
+	longitude.datatype = "float"
+
+	local altitude;
+	if show_altitude() then
+		altitude = s:option(Value, "altitude", site_i18n.translate("gluon-config-mode:altitude-label"), pkg_i18n.translatef("e.g. %s", "11.51"))
+		altitude.default = uci:get("gluon-node-info", location, "altitude")
+		altitude:depends(geolocation, "static")
+		altitude.datatype = "float"
+		altitude.optional = true
+	end
+	function geolocation:write(data)
+		if data == "static" then
+			uci:set("gluon-node-info", location, "latitude", latitude.data)
+			uci:set("gluon-node-info", location, "longitude", longitude.data)
+			if show_altitude() then
+				if altitude.data then
+					uci:set("gluon-node-info", location, "altitude", altitude.data)
+				else
+					uci:delete("gluon-node-info", location, "altitude")
+				end
```

Comment:
This is cargo-culting from luci.model.uci. With simple-uci, `uci:set(p, s, o, nil)` is equivalent to `uci:delete(p, s, o)`, so you usually don't need to check where a value is set before passing it to uci:set.